### PR TITLE
utils: add AST helper functions for editing formulae

### PR DESCRIPTION
**Pull request by SeekingMeaning**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?
- [x] Have you successfully run `brew man` locally and committed any changes?

-----

This PR adds two functions:

- `Utils::AST.replace_formula_stanza!`
- `Utils::AST.add_formula_stanza!`

These will be useful for gradually replacing regexes (which can sometimes be unreliable)

### CODE REVIEW

**Review by MikeMcQuaid — APPROVED** [review]
Very nice so far 👏🏻 . Some basic unit tests for these would be nice (when you've nailed down the functionality).

File: Library/Homebrew/utils/ast.rb [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+  # @api private
+  module AST
+    class << self
+      extend T::Sig
+
+      def replace_formula_stanza!(formula_contents, name:, replacement:, type: nil)
+        processed_source, body_node = process_formula(formula_contents)
+        children = body_node.begin_type? ? body_node.children.compact : [body_node]
+        stanza_node = children.find { |child| call_node_match?(child, name: name, type: type) }
+        odie "Could not find #{name} stanza!" if stanza_node.nil?
```
Comment (dawidd6) [inline Library/Homebrew/utils/ast.rb:16]: Maybe a `raise` would fit better here?

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Great work here @SeekingMeaning!

**Comment by dawidd6** [comment]
I agree, this is very cool. Nice job!

**Review by dawidd6 — APPROVED** [review]

**Pull request merged**